packages/jest-cli/src/runGlobalHook.js

```diff
@@ -27,24 +33,42 @@ export default ({
   }
 
   if (globalModulePaths.size > 0) {
-    return Promise.all(
-      Array.from(globalModulePaths).map(async modulePath => {
-        if (!modulePath) {
-          return null;
-        }
-
-        // $FlowFixMe
-        const globalModule = require(modulePath);
-
-        if (typeof globalModule !== 'function') {
-          throw new TypeError(
-            `${moduleName} file must export a function at ${modulePath}`,
-          );
-        }
-
-        return globalModule(globalConfig);
-      }),
-    );
+    return pEachSeries(Array.from(globalModulePaths), async modulePath => {
+      if (!modulePath) {
+        return;
+      }
+
+      const projectConfig =
+        allTests
+          .map(t => t.context.config)
+          .find(c => c[moduleName] === modulePath) ||
+        // Fallback to first one
+        allTests[0].context.config;
```

Comment:
What about in case where there's no tests? wouldn't this fail?

Reply:
We never run get this far if there's no tests: https://github.com/facebook/jest/blob/e38e3680f5aa991d08ce95eaebc494a4b9b5da78/packages/jest-cli/src/runJest.js#L218-L243

Reply:
I lied 😅 #7745

Reply:
😂

Reply:
Immediately thought of this when I saw that issue too 🤣